Library/Homebrew/cmd/list.rb

```diff
@@ -20,13 +20,28 @@
 
 require "metafiles"
 require "formula"
+require "cli_parser"
 
 module Homebrew
   module_function
 
   def list
+    @args = CLI::Parser.parse do
+      switch "--unbrewed"
+      switch "--pinned"
+      switch "--versions"
+      switch "--full-name"
+      switch "--multiple"
+      switch :verbose
+    end
+
+    if @args.multiple? && !@args.versions?
```

Comment:
This if block can be removed. 
Specifying constraints on options is now possible. You can have a look at `cli_parser_spec.rb`  to look at corresponding test cases.

Reply:
do I need to add anything here? If so, what is the difference between `required_for` and `depends_on` (except for expressing opposite direction of a relationship) ?